dev/com.ibm.ws.jaxrs.2.0.client/src/com/ibm/ws/jaxrs20/client/security/LibertyJaxRsClientSSLOutInterceptor.java

```diff
@@ -71,36 +80,37 @@ public void handleMessage(Message message) throws Fault {
         // getSocketFactory will return null if either the ssl feature is not enabled
         // or if it is enabled but there is no SSL configuration defined.  A null here
         // means to use the JDK's SSL implementation.
-        if (getSocketFactory(sslRef) != null) {
+        SSLSocketFactory socketFactory = getSocketFactory(sslRef);
+        if (socketFactory != null) {
             Object disableCNCheckObj = message.get(JAXRSClientConstants.DISABLE_CN_CHECK);
             Conduit cd = message.getExchange().getConduit(message);
-            configClientSSL(cd, sslRef, PropertyUtils.isTrue(disableCNCheckObj));
+            configClientSSL(cd, sslRef, PropertyUtils.isTrue(disableCNCheckObj), socketFactory);
         }
     }
 
-    private void configClientSSL(Conduit conduit, String sslRef, boolean disableCNCheck) {
+    private void configClientSSL(Conduit conduit, String sslRef, boolean disableCNCheck, SSLSocketFactory socketFactory) {
 
         //for HTTPS protocol
         if (conduit instanceof HTTPConduit) {
             HTTPConduit httpConduit = (HTTPConduit) conduit;
 
-            TLSClientParameters tlsClientParams = retriveHTTPTLSClientParametersUsingSSLRef(httpConduit, sslRef, disableCNCheck);
-            if (null != tlsClientParams) {
-                httpConduit.setTlsClientParameters(tlsClientParams);
+            if (null == httpConduit.getTlsClientParameters() || overrideUserTLS) {
```

Comment:
just a comment on the style, I don't normally see ```null != var``` it's usually ```var != null```
not that it matters